src/ray/gcs/redis_context.cc

```diff
@@ -46,79 +46,71 @@ namespace ray {
 
 namespace gcs {
 
-CallbackReply::CallbackReply(redisReply *redis_reply) {
+CallbackReply::CallbackReply(redisReply *redis_reply) : reply_type_(redis_reply->type) {
   RAY_CHECK(nullptr != redis_reply);
-  RAY_CHECK(redis_reply->type != REDIS_REPLY_ERROR)
-      << "Got an error in redis reply: " << redis_reply->str;
-  this->redis_reply_ = redis_reply;
+
+  switch (reply_type_) {
```

Comment:
How about add a private function ParseRedisReply() to make it more clearer.

Reply:
I would generally be in favor but given that the constructor doesn't do anything else I'd rather leave it there for simplicity.